src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/prison/PrisonApiClient.kt

```diff
@@ -300,10 +300,13 @@ class PrisonApiClient(
       .uri(
         "api/movements/$agencyId/in?fromDateTime=" +
           "${fromDate.format(DateTimeFormatter.ISO_LOCAL_DATE_TIME)}&toDateTime=" +
-          "${toDate.format(DateTimeFormatter.ISO_LOCAL_DATE_TIME)}",
+          toDate.format(DateTimeFormatter.ISO_LOCAL_DATE_TIME),
       )
       .header("Page-Limit", "10000")
       .retrieve()
       .bodyToMono(typeReference<List<Movement>>())
+      .onErrorResume(WebClientResponseException::class.java) {
+        Mono.just(emptyList())
+      }
       .block() ?: emptyList()
```

Comment:
Either we trigger a solely WebClientResponseException and then we handle it 4xx server error  , 404 etc.
Or we deliver an emptyList()

But i think original code doesn't deliver 404 when it gets 404 from Prison-API.

Reply:
The code will return 500 error 
![Screenshot 2024-09-04 at 13 05 27](https://github.com/user-attachments/assets/0635e30b-107c-4fba-8e98-8f1d29a187ab) .
I do not recommend returning the empty list as this means it is no record for prison XXX. But the reality is the prison XXX not exist. 
If you want to return correct error code I recon to follow this example https://github.com/ministryofjustice/hmpps-welcome-people-into-prison-api/blob/04e77e08aaf4269588aae3656bd61cd9027ae6c3/src/main/kotlin/uk/gov/justice/digital/hmpps/bodyscan/apiclient/BodyScanPrisonApiClient.kt#L41  
It is also good to check in all clients consuming `https://welcome-api-dev.prison.service.justice.gov.uk/prisons/LDF/recent-arrivals` how the 500 and 404 error is communicated to the users.

Reply:
@michalnawrockidj 
this was the PRISON-API we tested yesterday,
agencyID = XXX 
{
    "status": 404,
    "userMessage": "Resource with id [XXX] not found.",
    "developerMessage": "Resource with id [XXX] not found."
}
So the code is catching 404 and _I have swap code from emptyList to trigger 404 error.

Reply:
After discussion/chat with @michalnawrockidj 
A) As WPIP-UI (Front End) and some consumers might use 500 and We discuss couple of test on the top.
B) I will remove throwing 404 and I will capture the scenario discussed previously